map_location hasn't been implemented in load_from_checkpoint

**Describe the bug**
Current `load_from_checkpoint` function does not support to load a gpu checkpoint on cpu devices, i.e. map the location. 

**To Reproduce**
```
# on a cpu only device:
Model.load_from_checkpoint("GPU_CHECKPOINT_FILE")
```

**Expected behavior**
`AssertionError: Torch not compiled with CUDA enabled`

**Desktop (please complete the following information):**
 - OS: macOS Catalina 10.15.1
 - Browser: chrome
 - Version: 0.5.3.1

--- a/pytorch_lightning/core/lightning.py
+++ b/pytorch_lightning/core/lightning.py
@@ -930,7 +930,7 @@ class LightningModule(GradInformation, ModelIO, ModelHooks):
         return None
 
     @classmethod
-    def load_from_metrics(cls, weights_path, tags_csv):
+    def load_from_metrics(cls, weights_path, tags_csv, map_location=None):
         """Primary way of loading model from csv weights path.
 
         :param str weights_path: Path to a PyTorch checkpoint
@@ -975,9 +975,10 @@ class LightningModule(GradInformation, ModelIO, ModelHooks):
         hparams = load_hparams_from_tags_csv(tags_csv)
         hparams.__setattr__('on_gpu', False)
 
-        # load on CPU only to avoid OOM issues
-        # then its up to user to put back on GPUs
-        checkpoint = torch.load(weights_path, map_location=lambda storage, loc: storage)
+        if map_location is not None:
+            checkpoint = torch.load(weights_path, map_location=map_location)
+        else:
+            checkpoint = torch.load(weights_path, map_location=lambda storage, loc: storage)
 
         # load the state_dict on the model automatically
         model = cls(hparams)
@@ -989,7 +990,7 @@ class LightningModule(GradInformation, ModelIO, ModelHooks):
         return model
 
     @classmethod
-    def load_from_checkpoint(cls, checkpoint_path):
+    def load_from_checkpoint(cls, checkpoint_path, map_location=None):
         """
         Primary way of loading model from a checkpoint
         :param checkpoint_path:
@@ -997,9 +998,11 @@ class LightningModule(GradInformation, ModelIO, ModelHooks):
         :return:
         """
 
-        # load on CPU only to avoid OOM issues
-        # then its up to user to put back on GPUs
-        checkpoint = torch.load(checkpoint_path, map_location=lambda storage, loc: storage)
+        if map_location is not None:
+            checkpoint = torch.load(checkpoint_path, map_location=map_location)
+        else:
+            checkpoint = torch.load(checkpoint_path, map_location=lambda storage, loc: storage)
+
         try:
             ckpt_hparams = checkpoint['hparams']
         except KeyError:

